### PR TITLE
Code restructure and new unit tests

### PR DESCRIPTION
**Pull request by mbengit**

### CODE REVIEW

File: qpylib/abstract_qpylib.py [hunk 1]
```diff
@@ -120,48 +70,88 @@ def map_log_level(self, log_level='INFO'):
...
+        if relative_path == '':
+            return os.path.join(self._root_path(), 'store')
+        return os.path.join(self._root_path(), 'store', relative_path)
+
+    def get_root_path(self, relative_path):
+        return os.path.join(self._root_path(), relative_path)
 
     @abstractmethod
-    def get_console_address(self):
+    def _root_path(self):
```
Comment (wilhaddenIBM) [inline qpylib/abstract_qpylib.py]: root_path was only ever there for SDK support, do we still need it?

File: qpylib/abstract_qpylib.py [hunk 2]
```diff
@@ -171,6 +161,9 @@ def to_json_dict(self, python_obj, classkey=None):
         @return dict object containing key:value pairs for the python
         objects fields. Useable with JSON REST.
         """
+        print(type(python_obj))
```
Comment (jthomperoo) [inline qpylib/abstract_qpylib.py]: Why is the object being printed?

File: qpylib/live_qpylib.py [hunk 3]
```diff
@@ -9,102 +9,101 @@
...
-            headers = {}
+    def get_app_id(self):
+        return self._get_manifest_field_value('app_id', 0)
 
-        if version is not None:
-            headers['Version'] = version
+    def get_app_name(self):
+        return self._get_manifest_field_value('name')
+    
+    def _get_manifest_location(self):
```
Comment (wilhaddenIBM) [inline qpylib/live_qpylib.py]: Not sure if this is necessary as it was done for the SDK

File: qpylib/live_qpylib.py [hunk 4]
```diff
@@ -9,102 +9,101 @@
...
+        return self.APP_MANIFEST_LOCATION
 
-        if headers.get('Host') is None:
-            headers['Host'] = gethostbyname(gethostname())
+    def _root_path(self):
```
Comment (wilhaddenIBM) [inline qpylib/live_qpylib.py]: Also done for SDK

File: qpylib/abstract_qpylib.py (hunk 1 above)
Comment (mbengit) [reply qpylib/abstract_qpylib.py]: Probably not. Next step is to move/remove SDK-specific stuff.

File: qpylib/live_qpylib.py (hunk 3 above)
Comment (mbengit) [reply qpylib/live_qpylib.py]: I'll check and remove it if not necessary.

File: qpylib/abstract_qpylib.py (hunk 2 above)
Comment (mbengit) [reply qpylib/abstract_qpylib.py]: Good spot. It was there to debug a python 3 issue. I'll remove it.

File: qpylib/live_qpylib.py (hunk 4 above)
Comment (mbengit) [reply qpylib/live_qpylib.py]: As above.

**Pull request merged**